app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -305,6 +318,21 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
                 "finishing" to isFinishing.toString()
             )
         )
+
+        startupTelemetryOnStopCalled()
+    }
+
+    private fun startupTelemetryOnStopCalled() {
+        // remove predraw listener for 2 reasons
+        // 1. the application is shutting down and we want to avoid memory leak.
+        // 2. in case the application is also going in the background (instead of just the activity),
+        // we need to stop listening to the preDraw listener because if we have onPreDraw listener
+        // attached and the application comes foreground again, the activity will invoke onPreDraw()
+        // before onRestart and we want to avoid that.
+        rootContainer.viewTreeObserver.removeOnPreDrawListener(components.appStartupTelemetry)
+
+        // reset the startup time to avoid any weird state.
+        components.appStartupTelemetry.resetStartupTime()
```

Comment:
I think the comment is kinda redundant the name is self explanatory 😄

Reply:
I guess I was trying to explain as to why im reseting it.